Common/Arm64Emitter.cpp

```diff
@@ -3918,17 +4033,32 @@ void ARM64FloatEmitter::MOVI2F(ARM64Reg Rd, float value, ARM64Reg scratch, bool
 }
 
 // TODO: Quite a few values could be generated easily using the MOVI instruction and friends.
```

Comment:
outdated todo now

Reply:
Actually, there are more but I wasn't sure if there's impacts to perf to mix MOVI and float ops.  Probably not.  Still would have to find the constants.

ARM64 supports to vector constant types: one is a packed, 8-bit float with a very small exponent.  The other is a simple bitpattern, where you can create basically any of these, where AA is your own defined constant (must be aligned to 8 bit):
 * 0xAAAAAAAAAAAAAAAA
 * 0x00AA00AA00AA00AA
 * 0xAA00AA00AA00AA00
 * 0x000000AA000000AA
 * 0x0000AA000000AA00
 * 0x00AA00000x00AA0000
 * 0x0000AAFF0000AAFF
 * 0x00AAFFFF00AAFFFF

And then the inversion of all of those (i.e. 0xFF55FF55FF55FF55 etc.)  Lastly, you can have a constant that expands an 8-bit constant into 8 bytes of 00 or FF.

These generated 64-bit constant options are then doubled to both 64-bit lanes of a Q, or loaded to the lower lane with the upper lane zeroed (as per all other D vector operations.)

So it can actually do a lot, pretty useful for example for a BSL constant, and might maybe be better for some Vec4Blend (but only when top and bottom 64-bits of the desired BSL mask match, and there's no dest/src1 overlap.)  Anyway, these constants aren't easy to represent as a `float` arg anyway so I left it as just MOVI for now.

-[Unknown]

Reply:
Ah okay, makes sense. Thanks for the overview!